02_calculator/calculator.rb

```diff
@@ -1 +1,27 @@
-#write your code here
+def add(a, b)
+    a + b
+end
+
+def subtract(a, b)
+    a - b
+end
+
+def sum(arr)  
+    if(arr.length <= 0)
+        return 0
+    end
+    return arr.inject { |memo, n| memo + n }      
+end
+
+def multiply(*numbers)
+    numbers.reduce(1) { |memo, n| memo * n }
+end
+
+def power(base, exponent)
+    base ** exponent
+end
+
+def factorial(n)
+    (1..n).inject(:*) || 1
```

Comment:
Similar to above, you don't need to handle a special case if you provided an initial value:

```suggestion
    (1..n).inject(1, :*)
```

Additionally, you can save one operation because any number multiplied by 1 is the same number.

```suggestion
    (2..n).inject(:*) || 1
```
```